Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Queries/FullEntityBuilders.hs

```diff
@@ -45,18 +52,80 @@ import Storage.Tabular.FarePolicy
 import qualified Storage.Tabular.FarePolicy as FarePolicy
 import qualified Storage.Tabular.FarePolicy.FarePolicyProgressiveDetails as FarePolicyProgressiveDetails
 import qualified Storage.Tabular.FarePolicy.Instances as FarePolicy
+import Storage.Tabular.Location
+import Storage.Tabular.LocationMapping
 import qualified Storage.Tabular.QuoteSpecialZone as QuoteSpecialZone
+import Storage.Tabular.Ride
+import Storage.Tabular.Ride.Instances
+import Storage.Tabular.SearchRequest
+import Storage.Tabular.SearchRequestSpecialZone
 
 buildFullBooking ::
   Transactionable m =>
   BookingT ->
   DTypeBuilder m (Maybe (SolidType FullBookingT))
-buildFullBooking bookingT@BookingT {..} = runMaybeT $ do
-  fromLocationT <- Esq.findByIdM @BookingLocationT fromLocationId
-  toLocationT <- Esq.findByIdM @BookingLocationT toLocationId
-  fareParamsT <- Esq.findByIdM @FareParams.FareParametersT fareParametersId
-  fullFareParamsData <- MaybeT $ getFullFareParamsData fareParamsT
-  return $ extractSolidType @Booking (bookingT, fromLocationT, toLocationT, fullFareParamsData)
+buildFullBooking bookingT@BookingT {..} = do
+  mappings <- findAllLocationMappingsByTagId id
+  let allIds :: [LocationTId] = map (\(LocationMappingT {locationId}) -> locationId) mappings
+  allLocations <- mapM (Esq.findById' @LocationT) allIds
+  let x = sequence allLocations
+
+  case x of
+    Just locations -> do
+      fromLoc <- headMay locations & fromMaybeM (InternalError "From location does not exist.")
+      booking <- runMaybeT $ do
+        fareParamsT <- MaybeT $ Esq.findById' @FareParams.FareParametersT fareParametersId
+        fullFareParamsData <- MaybeT $ getFullFareParamsData fareParamsT
+        return (bookingT, fromLoc, drop 1 locations, fullFareParamsData)
+      case booking of
+        Just book -> return $ Just (extractSolidType @Booking book)
+        Nothing -> return Nothing
+    Nothing -> return Nothing
```

Comment:
This function written partially in `MaybeT` transformer. We can refactor it and use `MaybeT` everywhere. When at least computation in `MaybeT` will fail, we will have `Nothing` as result:

```
buildFullBooking ::
  Transactionable m =>
  BookingT ->
  DTypeBuilder m (Maybe (SolidType FullBookingT))
buildFullBooking bookingT@BookingT {..} = runMaybeT $ do
  mappings <- lift $ findAllLocationMappingsByTagId id
  let allIds :: [LocationTId] = map (\(LocationMappingT {locationId}) -> locationId) mappings
  allLocations <- mapM (Esq.findByIdM @LocationT) allIds
  fromLoc <- hoistMaybe $ headMay allLocations
  fareParamsT <- Esq.findByIdM @FareParams.FareParametersT fareParametersId
  fullFareParamsData <- MaybeT $ getFullFareParamsData fareParamsT
  let fullBookingT = (bookingT, fromLoc, drop 1 allLocations, fullFareParamsData)
  return (extractSolidType @Booking fullBookingT)
```